### PR TITLE
Improve admin page

### PR DESCRIPTION
**Pull request by chrisjwelly**
In this PR, I have made the following changes:
* Changed all the links to be buttons. This makes it clear whether a text is clickable or not
* Integrated the [QuillJS](https://github.com/quilljs/quill/) editor (with reference from this [article](https://stackoverflow.com/questions/42967325/how-to-capture-content-typed-in-a-rich-text-editor-and-save-it-in-a-rails-model) and this StackOverflow [post](https://stackoverflow.com/a/53155987/11284745)

Note: There is a need to sanitise content from Quill editor like this: 
`<%= raw sanitize @event.event_info.description, tags: %w(strong em a br), attributes: %w(href) %>`,
with the appropriate HTML tags such as `strong`, `em`, `br`.
If this is not done, the content will be rendered with the HTML elements. E.g. Instead of showing '**bolded text**', it shows '`<strong>bolded text</strong>`'

Also, we can probably add more tooltips for the editors.

Lastly, there are three themes: snow, core, and bubble. I have added the 3 css files to the project for convenience of switching, but in reality we may only need to stick to one.

### CODE REVIEW

**Comment by chrisjwelly** [comment]
@stevenwjy I have added the quill text editor to only the full description of events. Please have a look if it's ok! 

I don't think all the fields will require a WYSIWYG editor; can you advise which fields would be appropriate for this? I will then make more changes. 

Also, there seems to be a lot of file additions by integrating quill :sweat_smile:
Edit: Changed to quill minified

If you feel that it's better, I can separate the integration of quill from the changing links to become buttons.

**Review by stevenwjy — DISMISSED** [review]

File: app/views/admin/events/_form.html.erb [hunk 1]
```diff
@@ -101,6 +100,9 @@
       theme: 'snow'  // or 'bubble'
     });
 
+    var postContentInput = document.querySelector('#event_event_info_attributes_description');
```
Comment (stevenwjy) [inline app/views/admin/events/_form.html.erb:126]: Where is this CSS id actually being set?

File: app/views/admin/events/show.html.erb [hunk 2]
```diff
@@ -31,7 +31,9 @@
         </div>
         <div class="row">
           <div class="col-lg-2 col-md-12 field-key">Description</div>
-          <div class="col-lg-10 col-md-12 field-value"><%= @event.event_info.description %></div>
+          <div class="col-lg-10 col-md-12 field-value">
+            <%= raw sanitize @event.event_info.description, tags: %w(strong em a br), attributes: %w(href) %>
```
Comment (stevenwjy) [inline app/views/admin/events/show.html.erb]: Here you only sanitize the bold, italic, newline, and link. Meanwhile in the text editor, there is actually no support for link (a href), and here you also don't sanitize the underline effect... probably can consider to update this part.

Also, when I tried the text editor, the box height kept increasing as I typed new line or add more text, can it be adjusted to just show a fixed height and scroll text if the box if not enough to contain all the content.

**Review by stevenwjy — CHANGES_REQUESTED** [review]
For the theme, I tried using 'snow', 'core', and 'bubble'.

The 'core' doesn't seem to work as it didn't show anything (?), here is the screenshot:
Screen Shot 2019-07-12 at 00 49 49

For the bubble, I don't know why but it doesn't look like the one in [here](https://quilljs.com/docs/themes/), here is the screenshot:
Screen Shot 2019-07-12 at 00 50 32

Anyway ... I don't think we will need those two, so can just ignore and remove them from the code, the theme 'snow' seems good enough for what we need right now.

However, it looks a bit weird here:
Screen Shot 2019-07-12 at 00 47 59

I think that it will be better if you can just set the background color to be white and add some padding to ensure that it doesn't touch the other input field.

Thank you!

File: app/views/admin/events/_form.html.erb (hunk 1 above)
Comment (stevenwjy) [reply app/views/admin/events/_form.html.erb:126]: Just asking because I don't really understand how it can work ahahaha

Comment (chrisjwelly) [reply app/views/admin/events/_form.html.erb:126]: I should have clarified!

During runtime, the code `<%= event_info_form.hidden_field :description, required: true %>
` gets resolved to have `id="event_event_info_attributes_description"`.

The following example shows the original view in the `development` (as of 7c6f100b0bed17c054e6ccb1a08a89079ca6851e). Which shows that `<%= event_info_form.text_area :description, required: true %>` has the `id="event_event_info_attributes_descrition"`


I agree that it is quite magical with rails doing its own job, but so far this is the best option I have found. During my testing in my own sandbox (unrelated to gapura) I had this code snippet with a Post model:
```ruby
<h1>Add Post</h1>
<%= form_for :post, url: posts_path, remote: true, html: { id: 'my-form' } do |f| %>
  <p>
    <%= f.label :title %><br>
    <%= f.text_field :title %>
  </p>

  <p>
    <%= f.label :body %><br>
    <div id="editor-container" hidden="true">   
      <%= f.text_area :body %>
    </div>
    <div id="editor"></div>
  </p>

  <p>
    <%= f.submit %>
  </p>
<% end %>
```

and I found that ` var postContentInput = document.querySelector('#post_body');` referred to the line `<%= f.text_area :body %>`.

File: app/views/admin/events/show.html.erb (hunk 2 above)
Comment (chrisjwelly) [reply app/views/admin/events/show.html.erb]: Noted!

**Comment by chrisjwelly** [comment]
@stevenwjy About the themes, I just realised about that too. I initially thought this was the offending code, because I only added the snow theme to the asset pipeline:
https://github.com/nuansa-cp/gapura/blob/8a697dbc3ee51d9940e675e3aa99299a318a9c29/app/views/layouts/application.html.erb#L44

But upon adding the two themes in the asset pipeline, I experienced the same thing as you. I will remove it.

File: app/views/admin/events/_form.html.erb (hunk 1 above)
Comment (stevenwjy) [reply app/views/admin/events/_form.html.erb:126]: I see, okok thanks for the clarif hahahha

Comment (chrisjwelly) [reply app/views/admin/events/_form.html.erb:126]: I think I should add this under comments in the code for future maintainers. Currently I’m outside; will add it tomorrow!

File: lib/assets/stylesheets/quill/quill.snow.css [hunk 3]
```diff
@@ -850,6 +850,7 @@
   box-sizing: border-box;
   font-family: 'Helvetica Neue', 'Helvetica', 'Arial', sans-serif;
   padding: 8px;
+  background: white;
```
Comment (stevenwjy) [inline lib/assets/stylesheets/quill/quill.snow.css]: Please never edit the original code in the lib folder unless there is no other choice, if you want to add extra feature like this, can just add it to CSS file in the app/assets

**Review by stevenwjy — APPROVED** [review]

File: app/views/admin/events/_form.html.erb (hunk 1 above)
Comment (stevenwjy) [reply app/views/admin/events/_form.html.erb:126]: Okok, no problem, I have added it :)

**Pull request merged**